### ISSUE TITLE
[Feature request] Make Excalidraw compatible with Svelte.

### ISSUE THREAD

**Issue by ankified** [excalidraw/excalidraw#6718]
I'm learning to code in order to create a website and I decided to make it in Svelte as it seems to be more "beginner friendly". I intend to integrete Excalidraw into this website, however, I've just skimmed through the docs and noticed it uses React. Are there any chances that there's a version of Excalidraw compatible with Svelte in the future as well?

**Comment by dev10110** [comment]
any chance this has been addressed?

**Comment by SrZorro** [comment]
At our internal framework we have Excalidraw/React working inside Svelte

You can check a minimal implementation in the following REPL: https://www.sveltelab.dev/yg89k40qsw1uufc

There is a ton of features missing (save to cloud, external images, sync multiplayer...), we have some of them already implemented in our framework, but its not easy to make a quick REPL with the server features, but that REPL its a good starting point

**Comment by ankified** [comment]
@SrZorro Thank you. That's really interesting. I didn't know it's possible to run React inside Svelte. Impressive!

**Comment by tokuhirom** [comment]
@SrZorro 
Thank you for the great sample code. But it looks like the setAPI method is never called.
Why?

**Comment by tokuhirom** [comment]
In my environment, the code works if i use `excalidrawAPI={setAPI}` instead of `ref={setAPI}.

**Comment by SrZorro** [comment]
> Thank you for the great sample code. But it looks like the setAPI method is never called.
> Why?

You are right its not getting called. When I saw the issue I did a quick copy to sveltelab of what I had working and has some unfinished code

The objective of `setAPI` was to expose the `ExcalidrawImperativeAPI` to the consumer of the `Excalidraw.svelte` component via the `bind:excalidrawAPI` at `AsyncExcalidraw.svelte` but did not finished the `ReactComponent.svelte` side to finish it.

**Comment by fabhari** [comment]
loading react library into svelte is increasing the package bundle size around 2mb, which is so huge and breaks the purpose of migrating my project to svelte  👎 , is there any community working on building excalidraw or any alternative built in pure svelte ?
